extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/extractors/SoundcloudStreamExtractor.java

```diff
@@ -286,10 +274,42 @@ public String getHlsUrl() {
         return audioStreams;
     }
 
-    private static String urlEncode(String value) {
+    private final static String PATTERN_WEB_URLS_IN_HLS_MANIFESTS = "((http?|https?):((//)|(\\\\))+[\\w\\d:#@%/;$()~_?+-=\\\\.&]*)";
```

Comment:
Please make the compiled pattern static, not the string. Also (http?|https?) => (https?)? i think

Reply:
I think all SoundCloud streaming formats are https but if they are http, the extractor can extract them.